### PR TITLE
Colinanderson/play kml tour compose

### PR DESCRIPTION
**Pull request by colinanderson**
## Description

Change the play KML tour sample to a compose implementation

## Links and Data

issue: https://devtopia.esri.com/runtime/kotlin/issues/4995
- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/sampleviewer/) Job for this PR has been run
  - [x] link: https://runtime-kotlin.esri.com/view/all/job/vtest/job/sampleviewer/58/

## What To Review
- Review the code to make sure it is easy to follow like other samples on Android
- `README.md` and `README.metadata.json` files

## How to Test
Run the sample on the sample viewer or the repo.

### CODE REVIEW

File: samples/play-kml-tour/src/main/java/com/esri/arcgismaps/sample/playkmltour/components/PlayKMLTourViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        elevationSources.add(ArcGISTiledElevationSource("https://elevation3d.arcgis.com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"))
+    }
+
+    // create a KML layer from a KML dataset with a KML tour
+    private val kmlDataSet = KmlDataset(provisionPath + File.separator + "Esri_tour.kmz")
+    private val kmlLayer = KmlLayer(kmlDataSet)
+
+    // create a scene with the surface and KML layer
+    val arcGISScene =
+        ArcGISScene(BasemapStyle.ArcGISImagery).apply {
```
Comment (TADraeseke) [inline samples/play-kml-tour/src/main/java/com/esri/arcgismaps/sample/playkmltour/components/PlayKMLTourViewModel.kt:64]: can probably bump this up a line

File: samples/play-kml-tour/src/main/java/com/esri/arcgismaps/sample/playkmltour/components/PlayKMLTourViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            operationalLayers.add(kmlLayer)
+        }
+
+    val sceneViewProxy = SceneViewProxy()
+
+    private var kmlTour: KmlTour? = null
+    private val kmlTourController = KmlTourController()
+
+    private val _statusFlow: MutableStateFlow<KmlTourStatus> = MutableStateFlow(KmlTourStatus.NotInitialized)
+    val statusFlow = _statusFlow.asStateFlow()
```
Comment (TADraeseke) [inline samples/play-kml-tour/src/main/java/com/esri/arcgismaps/sample/playkmltour/components/PlayKMLTourViewModel.kt:76]: could these maybe more explictly named? This code gets read outside of an IDE on the website, so more verbose names can be helpful

_tourStatus or _tourStatusFlow maybe?

File: samples/play-kml-tour/src/main/java/com/esri/arcgismaps/sample/playkmltour/screens/PlayKMLTourScreen.kt [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+                LinearProgressIndicator(progress = { progressFlow },
+                    modifier = Modifier.fillMaxWidth().padding(all = padding))
+                Text("${stringResource(R.string.tour_status)} ${tourStateToString(stateFlow)}",
+                    modifier = Modifier.padding(all = padding).align(Alignment.CenterHorizontally))
+                Row(modifier = Modifier.fillMaxWidth()) {
+                    Button(enabled = (stateFlow == KmlTourStatus.Paused || stateFlow == KmlTourStatus.Playing),
+                        modifier = Modifier.fillMaxWidth(0.5f).padding(all = padding), onClick = {
+                        viewModel.reset()
+                    }) {
```
Comment (TADraeseke) [inline samples/play-kml-tour/src/main/java/com/esri/arcgismaps/sample/playkmltour/screens/PlayKMLTourScreen.kt:86]: We normally make the chain of modifer.padding.fill.etc.etc vertically so more like this:

```suggestion
                LinearProgressIndicator(
                progress = { progressFlow }, modifier = Modifier
                    .fillMaxWidth()
                    .padding(all = padding)
            )
            Text(
                "${stringResource(R.string.tour_status)} ${tourStateToString(stateFlow)}",
                modifier = Modifier
                    .padding(all = padding)
                    .align(Alignment.CenterHorizontally)
            )
            Row(
                modifier = Modifier.fillMaxWidth()
            ) {
                Button(enabled = (stateFlow == KmlTourStatus.Paused || stateFlow == KmlTourStatus.Playing),
                    modifier = Modifier
                        .fillMaxWidth(0.5f)
                        .padding(all = padding),
                    onClick = {
                        viewModel.reset()
                    }) {
```

**Review by TADraeseke — CHANGES_REQUESTED** [review]
Looking good @colinanderson -- just some small things below

**Comment by colinanderson** [comment]
@TADraeseke that should be all your comments addressed.

**Review by TADraeseke — APPROVED** [review]
LGTM!

**Comment by colinanderson** [comment]
Updated the enable/disable of the play button to disable when the tour is not initialised. Also renamed classes from `KML` to` Kml` and some other renames based on https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/pull/287#discussion_r1871864445.

New vTest - https://runtime-kotlin.esri.com/view/all/job/vtest/job/sampleviewer/62

**Review by 01smito01 — APPROVED** [review]
Looks good, just a few suggestions.

File: samples/play-kml-tour/README.metadata.json [hunk 4]
```diff
@@ -31,8 +33,10 @@
...
     "snippets": [
+        "src/main/java/com/esri/arcgismaps/sample/playkmltour/components/PlayKMLTourViewModel.kt",
+        "src/main/java/com/esri/arcgismaps/sample/playkmltour/DownloadActivity.kt",
         "src/main/java/com/esri/arcgismaps/sample/playkmltour/MainActivity.kt",
-        "src/main/java/com/esri/arcgismaps/sample/playkmltour/DownloadActivity.kt"
+        "src/main/java/com/esri/arcgismaps/sample/playkmltour/screens/PlayKMLTourScreen.kt"
     ],
     "title": "Play KML tour"
-}
+}
```
[the remark this answers is not in the document]
Comment (colinanderson) [reply samples/play-kml-tour/README.metadata.json]: Added

File: samples/play-kml-tour/src/main/java/com/esri/arcgismaps/sample/playkmltour/components/PlayKmlTourViewModel.kt [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+                )
+            }
+            kmlLayer.load().onSuccess {
+                kmlTour = findFirstKMLTour(kmlDataSet.rootNodes)
+                kmlTourController.tour = kmlTour
+
+                if (kmlTour != null) {
+                    collectKmlTourStatus(kmlTour!!)
+                    collectProgress(kmlTourController)
+                }
```
[the remark this answers is not in the document]
Comment (colinanderson) [reply samples/play-kml-tour/src/main/java/com/esri/arcgismaps/sample/playkmltour/components/PlayKmlTourViewModel.kt]: Changed this and another `if` to be more Kotlin style.

File: samples/play-kml-tour/src/main/java/com/esri/arcgismaps/sample/playkmltour/screens/PlayKmlTourScreen.kt [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+import com.arcgismaps.mapping.kml.KmlTourStatus
+import com.arcgismaps.toolkit.geoviewcompose.SceneView
+import com.esri.arcgismaps.sample.playkmltour.R
+import com.esri.arcgismaps.sample.playkmltour.components.PlayKmlTourViewModel
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialog
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
```
[the remark this answers is not in the document]
Comment (colinanderson) [reply samples/play-kml-tour/src/main/java/com/esri/arcgismaps/sample/playkmltour/screens/PlayKmlTourScreen.kt]: Done

File: samples/play-kml-tour/src/main/java/com/esri/arcgismaps/sample/playkmltour/screens/PlayKmlTourScreen.kt [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+ */
+@Composable
+fun PlayKmlTourScreen(sampleName: String) {
+    val viewModel: PlayKmlTourViewModel = viewModel()
+
+    val kmlTourStatusFlow = remember { viewModel.kmlTourStatusFlow }
+    val kmlTourStatus by kmlTourStatusFlow.collectAsStateWithLifecycle(KmlTourStatus.NotInitialized)
+
+    val kmlTourProgressFlow = remember { viewModel.kmlTourProgressFlow }
+    val kmlTourProgress by kmlTourProgressFlow.collectAsStateWithLifecycle(0.0f)
```
[the remark this answers is not in the document]
Comment (colinanderson) [reply samples/play-kml-tour/src/main/java/com/esri/arcgismaps/sample/playkmltour/screens/PlayKmlTourScreen.kt]: Changed as discussed.

**Comment by colinanderson** [comment]
@01smito01 I made some updates so please take another look - interested if the `let` usages are an improvement or not.

**Review by 01smito01 — APPROVED** [review]
I'm ambivalent on whether the `let`s are improvements, but I like the `when`. Everything else looks fine.

**Pull request merged**